runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -73,6 +73,343 @@ extern "C" void _patchVirtualGuard(uint8_t *locationAddr, uint8_t *destinationAd
 extern "C" void ASM_CALL _patchVirtualGuard(uint8_t*, uint8_t*, uint32_t);
 #endif
 
+// START OF BINARY TEMPLATES
+
+// These *BinaryTemplate structs describe the shape of the binary relocation records.
+struct TR_RelocationRecordBinaryTemplate
+   {
+   uint8_t type(TR_RelocationTarget *reloTarget);
+
+   uint16_t _size;
+   uint8_t _type;
+   uint8_t _flags;
+
+#if defined(TR_HOST_64BIT)
+   uint32_t _extra;
+#endif
+   };
+
+// Generating 32-bit code on a 64-bit machine or vice-versa won't work because the alignment won't
+// be right.  Relying on inheritance in the structures here means padding is automatically inserted
+// at each inheritance boundary: this inserted padding won't match across 32-bit and 64-bit platforms.
+// Making as many fields as possible UDATA should minimize the differences and gives the most freedom
+// in the hierarchy of binary relocation record structures, but the header definitely has an inheritance
+// boundary at offset 4B.
```

Comment:
maybe add a comment on each of the (currently) three structs listing the other two of the three BinaryTemplate structs that *must* all be identical in their first three (at least) fields.

I know you're just moving code here, so those comments could be added as a follow-up.

Reply:
Done in https://github.com/eclipse/openj9/pull/11235/commits/54e35a963015dcff3022ef2bf8a1daca301447c5. I also opened https://github.com/eclipse/openj9/issues/11659 to track further clean up of the binary templates, since we no longer need to worry about padding and what not.